toptal/picasso
# feat: refactor highlighting and add description for dropdown

[TEA-1521]
[TEA-1571]

### Description
- Decouple select and highlight for Select and Autocomplete options
- Add checkmark for selected options according to the [latest design](https://app.abstract.com/share/b2f57088-6713-47f1-a4fd-2f3f60a9d648?mode=design&sha=ebb7cf6ef9c38f840e79addb88810916c60be421)
- Refactor default highlighting for Select and Autocomplete
  - if only one option is selected, highlight selected option
  - otherwise, if no or multiple options are selected, highlight the first option
- Add feature to show descriptions inside Select and Autocomplete options

### How to test
Run `yarn start` and make sure Select and Autocomplete are working as described above

| Single Select                                 | Multiple Select                                  | Autocomplete |
| --------------------------------------- | --------------------------------------- |--------------------------- |
| ![image](https://user-images.githubusercontent.com/4954380/89761374-0ea6e600-db10-11ea-9c7d-4f590a7924e9.png) | ![image](https://user-images.githubusercontent.com/4954380/89761386-14043080-db10-11ea-9e3d-c88a9403274c.png) |  ![image](https://user-images.githubusercontent.com/4954380/89761417-267e6a00-db10-11ea-8172-e51241c71bda.png) |

| Select with description | Autocomplete with description |
| --------------------------------------- | --------------------------------------- |
| ![image](https://user-images.githubusercontent.com/4954380/89874403-9b6ba580-dbdd-11ea-9eec-69f7c56588d8.png) |  ![image](https://user-images.githubusercontent.com/4954380/89874381-973f8800-dbdd-11ea-8a5d-1e29d344ddee.png)|


### Review



[TEA-1521]: https://toptal-core.atlassian.net/browse/TEA-1521

[TEA-1571]: https://toptal-core.atlassian.net/browse/TEA-1571
